galasa-inttests-parent/dev.galasa.inttests/src/main/java/dev/galasa/inttests/compilation/AbstractCompilationLocalZip.java

```diff
@@ -0,0 +1,224 @@
+        Matcher matcher = pattern.matcher(fileData);
+        matcher.find();
+        // Insert the constraints closure between match 1 (dependencies) and match 2 (closing brace)
+        fileData = fileData.replace(matcher.group(0), matcher.group(1) + constraints + matcher.group(2));
+        
+        Files.write(fileToChange, fileData.getBytes());
+    }
+    
+    /*
+     * For use when changing source code to work with the isolated zip (either mvp or full).
+     * Specify a file to work against. This method will alter the selenium manager depedency
+     * to exclude several unnecessary packages that aren't available in the zip.
+     * 
+     * @param fileToChange the path to the file that needs updating
+     */
+    protected void addImplementationConstraints(Path fileToChange) throws IOException {
+        logger.info("Adding constraints (for selenium manager) to: " 
+                + fileToChange.getName(fileToChange.getNameCount()-2) + "/" + fileToChange.getFileName());
+        
+        String fileData = new String(Files.readAllBytes(fileToChange), Charset.defaultCharset());
+          
+        String constraints = 
+            "    implementation('dev.galasa:dev.galasa.selenium.manager:0.+'){\n" + 
+            "        exclude group: 'com.squareup.okio', module: 'okio'\n" + 
+            "        exclude group: 'com.squareup.okhttp3', module: 'okhttp'\n" + 
+            "        exclude group: 'net.bytebuddy', module: 'byte-buddy'\n" + 
+            "        exclude group: 'org.apache.commons', module: 'commons-exec'\n" + 
+            "        exclude group: 'com.google.guava', module: 'guava'\n" + 
+            "    }";
+        
+        // Regex Matches:
+        // Match 1: The dependencies closure, as well as whatever is inside it, up until just before the final, closing, curly brace.
+        // Match 2: The final, closing, curly brace.
+        String regex = "implementation\\s\\'dev\\.galasa\\:dev\\.galasa\\.selenium\\.manager\\:0\\.\\+\\'";
+        Pattern pattern = Pattern.compile(regex, Pattern.MULTILINE);
+        Matcher matcher = pattern.matcher(fileData);
+        matcher.find();
+        // Insert the constraints closure between match 1 (dependencies) and match 2 (closing brace)
+        fileData = fileData.replace(matcher.group(0), constraints);
+        
+        Files.write(fileToChange, fileData.getBytes());
+    }
+    
+    /*
+     * For use when changing source code to work with the isolated zip (either mvp or full).
+     * Specify a file to work against. This method will add all the dependencies specified in 
+     * the `dependencies` parameter, to the file specified (fileToChange)
+     * 
+     * @param fileToChange the path to the file that needs updating
+     * @param dependencies an array of manager names to be added (all must be within the dev.galasa group)
+     */
+    protected void addManagerDependencies(Path fileToChange, String[] dependencies) throws IOException {
```

Comment:
Not sure why we need this,  we need to test what is in the example build.gradle

Reply:
This adding (testing) of extra dependencies has been moved to a different test class.